### ISSUE TITLE
Suppport SdkResponseMetadata

### ISSUE THREAD

**Issue by zoewangg** [aws/aws-sdk-java-v2#670]
v1 supports ResponseMetadata to allow customers to retrieve useful debugging information such as requestId, hostId for the request. Currently, it's not implemented in v2.

**Comment by zoewangg** [comment]
For now, the requestIds might be retrieved from the response headers for some services.

For example:

s3: 
```java
        HeadObjectResponse response = s3.headObject(HeadObjectRequest.builder()
                                                                     .bucket(BUCKET_NAME)
                                                                     .key(key)
                                                                     .build());

        response.sdkHttpResponse().headers().get("x-amz-request-id");
        response.sdkHttpResponse().headers().get("x-amz-id-2");
```

DynamoDB:
```java
     result.sdkHttpResponse().headers().get("x-amzn-RequestId");
```

**Comment by zoewangg** [comment]
Implemented in #701
